### ISSUE TITLE
Why Iris-grib Windows support through conda?

### ISSUE THREAD

**Issue by geofbaum** [SciTools/iris-grib#94]
Iris-grib is available for Windows systems but it relies on gribapi and sooner or later ecCodes. However neither of those packages are available for a Windows environment with Python support? Am I missing something or is there a reason why you can load this package into a Windows environment even though it requires something that you cannot?

**Comment by djkirkham** [comment]
Hi geofbaum.

What channel are you getting it from? I'm unable to find it available for Windows on anaconda.org

**Comment by geofbaum** [comment]
It is listed here, https://anaconda.org/scitools/iris_grib

It seems to install fine just that it obviously throws the error looking for grib-api when you try to call it from a python console, notebook, etc.

**Comment by djkirkham** [comment]
I'm not quite sure why it's there, but it shouldn't be. Interestingly, the package doesn't contain any files to be deployed, so it's not clear to me why you are seeing an error about grib-api, I would have thought the error would be about iris_grib itself.

I should mention that the scitools channel is now deprecated; we are advising people to use conda-forge from now on.

**Comment by geofbaum** [comment]
Ah ok,  that's my bad then. I was on a search to find something that would integrate Grib files nicely with cartopy. Which is why I had hoped to use the grib-api  or something similar since my current work around is unfortunately converting them into netcdf files which I would prefer not to just to save space. Thanks for your help.

**Comment by geofbaum** [comment]
@djkirkham 
Here is the exact error that I was receiving just for future reference as people can install iris-grib directly from this repo to Windows without any issues as well as from the deprecated scitools channel. It's from the __init__.py file.

`ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-3-de95fb9ad1af> in <module>()
      1 import os
      2 import iris
----> 3 import iris_grib
      4 filename = "D:\\Downloads\\hrrr.t12z.wrfsfcf00.nc"
      5 cubes = iris_grib.load_cubes(filename)

D:\Anaconda3\lib\site-packages\iris_grib\__init__.py in <module>()
     32 import cartopy.crs as ccrs
     33 import cf_units
---> 34 import gribapi
     35 import numpy as np
     36 import numpy.ma as ma

ModuleNotFoundError: No module named 'gribapi'`

And this is the code that I was using that brought up the error.

`import os
import iris
import iris_grib
filename = "D:\\Downloads\\hrrr.t12z.wrfsfcf00.nc"
cubes = iris_grib.load_cubes(filename)
print(cubes)`

iris version 1.13.0 (installed from conda-forge)
iris-grib version 0.12.0 (installed from the github repo)

**Comment by Ruben-at-Github** [comment]
I ran into similar issues as described above.

Do I understand correctly that, on windows, it should be possible to install iris-grib as listed here: https://github.com/conda-forge/iris-grib-feedstock ?

If I try to do that then I get the error message that the package is not available from the https://conda.anaconda.org/conda-forge/win-64 channel.

I do need to read .grib files. Is there any way to solve this?

**Comment by pelson** [comment]
> Do I understand correctly that, on windows, it should be possible to install iris-grib (with conda-forge):

https://github.com/conda-forge/iris-grib-feedstock/blob/master/recipe/meta.yaml#L13 contains ```skip: True  # [win or py3k]``` which means that iris-grib isn't available on Windows, and isn't available on python 3.


> I do need to read .grib files. Is there any way to solve this?

Without ecCodes, we don't have the ability to read GRIB on Windows. The relevant issue on conda-forge https://github.com/conda-forge/eccodes-feedstock/issues/3 appears to have stalled somewhat.

Technically, I think it might be feasible (https://github.com/conda-forge/ecmwf_grib-feedstock/issues/5 has some positive information), but I suspect it will take a significant amount of effort to build ecCodes on Windows right now.

Truth is, I don't really have a suggested solution for GRIB with Iris on Windows.

**Comment by vciulei** [comment]
Hi @RubendeBruin @pelson @geofbaum  

Any updates on this matter? I've seen an Experimental Windows support on this matter here  : https://confluence.ecmwf.int//pages/viewpage.action?pageId=48109524 but since I am rather a noob, I can't tell if it works and how. 
Also this might be useful, although I am not sure it does what @pelson referred to: 

https://github.com/NMC-DAVE/dk_met_eccodes_tools

I am trying to read GRIB files with a win x64 , so far made a cmake build but got a critical error related to the OS - Win not supported. 

Any suggestions on how to read GRIB files  ( using ecCodes, Metview, ) and preferably python ( on Windows) are much much appreciated.

thanks

**Comment by geofbaum** [comment]
@vciulei I ended up using Unidata's ncl_convert2nc to convert the files that I was playing around with to netcdf format. Not really the result that I wanted but I haven't had much time to look into this since April. I believe I was close to getting it working using the Ubuntu on Windows app but I still believe I ran into some issues with that. Hopefully in the next few weeks when I'll have more time again after our harvest season is more finished I'll be able to recheck everything and post an update.
